api/v1/dois/PKPDoiHandler.php

```diff
@@ -269,28 +278,59 @@ public function edit(SlimRequest $slimRequest, APIResponse $response, array $arg
             return $response->withStatus(403)->withJsonError('api.dois.403.editItemOutOfContext');
         }
 
-        $params = $this->convertStringsToSchema(\PKP\services\PKPSchemaService::SCHEMA_DOI, $slimRequest->getParsedBody());
-        $params['id'] = $doi->getId();
+        $params = $this->convertStringsToSchema(PKPSchemaService::SCHEMA_DOI, $slimRequest->getParsedBody());
 
         $errors = Repo::doi()->validate($doi, $params);
         if (!empty($errors)) {
             return $response->withStatus(400)->withJson($errors);
         }
 
-        Repo::doi()->edit($doi, $params);
+        $pubObjectType = $slimRequest->getParsedBodyParam('pubObjectType');
+        $pubObjectId = $slimRequest->getParsedBodyParam('pubObjectId');
 
-        $doi = Repo::doi()->get($doi->getId());
+        // Default behaviour, only edits DOI
+        if (empty($pubObjectType) && empty($pubObjectId)) {
```

Comment:
I think we need to consider issue type here -- it always fits into this default behavior because there is no versioning for issues -- correct? Else, further down the type not found will be returned when an issue DOI is edited.

Reply:
Yes, I've rewritten it to account for issues in OJS so the issue type is now accounted for.